Xamarin.Build.AsyncTask/AsyncTask.cs

```diff
@@ -333,6 +354,11 @@ protected void WaitForCompletion()
                                 BuildEngine.LogCustomEvent(e);
                             }, customDataAvailable);
                             break;
+                        case WaitHandleIndex.TelemetryDataAvailable:
+                            LogInternal<TelemetryEventArgs>(telemetryMessageQueue, (e) => {
+                                BuildEngine5.LogTelemetry(e.EventName, e.Properties);
```

Comment:
Is there a way to know what version of MSBuild/VS they added `BuildEngine5`?

I think it would throw in older ones:

https://github.com/dotnet/msbuild/blob/6c748718699a44476ca68fd7ec424bae1bdab153/src/Utilities/Task.cs#L87

If it works in like VS 17.x, we are probably fine.

Reply:
I've set the minimum MSBuild package dependency to 17+ so this should not be a problem.